File: account-service/src/main/java/accountservice/exception/GeneralExceptionHandler.java
package accountservice.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@RestControllerAdvice
public class GeneralExceptionHandler extends ResponseEntityExceptionHandler {

    @ExceptionHandler(AccountNotFoundException.class)
    public ResponseEntity<Object> handleAccountNotFoundException(AccountNotFoundException ex) {
        return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
    }

    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<Object> handleNotFoundException(NotFoundException ex) {
        return new ResponseEntity<>(ex.getExceptionMessage(), HttpStatus.NOT_FOUND);
    }
}

File: review-service/src/main/kotlin/reviewservice/exception/GeneralExceptionHandler.kt
package reviewservice.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
import java.util.Objects

@RestControllerAdvice
class GeneralExceptionHandler : ResponseEntityExceptionHandler() {
    @ExceptionHandler(ReviewNotFoundException::class)
    fun handleReviewNotFoundException(ex: ReviewNotFoundException): ResponseEntity<Any> {
        return ResponseEntity.status(404).body(Objects.requireNonNull(ex.message))
    }

    @ExceptionHandler(NotFoundException::class)
    fun handleNotFoundException(ex: NotFoundException): ResponseEntity<Any> {
        return ResponseEntity.status(404).body(Objects.requireNonNull(ex.getMessage()))
    }
}